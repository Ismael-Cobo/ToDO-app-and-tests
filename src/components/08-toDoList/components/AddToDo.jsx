import { useState } from "react"

export const AddToDo = ({handleAddToDo}) => {
    
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 2) return

        const newTodo = {
            id: new Date().getTime(),
            desc: inputValue,
            done: false
        };

        handleAddToDo(newTodo)

        setInputValue('')
    } 


    return (
        <form className='todo__form' onSubmit={handleSubmit}>
            <label className='form__title' htmlFor='todo'> Today I need to </label>
            <div className='form__input__group'>
                <input
                    type='text'
                    id='todo'
                    className='form__input'
                    value={inputValue}
                    onChange={handleInputChange}

                />
                <input
                    type='submit'
                    value='submit'
                    className='form__btn'
                />
            </div>
        </form>
    )
}
