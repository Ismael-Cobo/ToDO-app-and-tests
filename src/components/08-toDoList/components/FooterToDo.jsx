import React from 'react'

export const FooterToDo = () => {
    return (
        <footer className='todo__footer'>
            Inspired in <a href='https://codepen.io/lokifullap/pen/gOXNGBo' target={'_blank'} rel="noreferrer">Codepen</a>
        </footer>
    )
}
