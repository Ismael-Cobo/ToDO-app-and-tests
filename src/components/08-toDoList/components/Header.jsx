import LaptopSVG from '../../../styles/img/laptop.svg'

export const Header = () => {
    return (
        <div className='todo__heading'>
            <img className='heading__img' src={LaptopSVG} alt='heading Img' />
            <h1 className='heading__title'>To-Do List</h1>
        </div>
    )
}
