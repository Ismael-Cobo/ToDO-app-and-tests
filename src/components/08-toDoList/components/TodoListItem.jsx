import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleToggle }) => {
    return (
        <>
            <li
                className={`todo__item ${todo.done ? 'todo__done' : ''}`}
            >
                <p
                    className='todo__item__text'
                    onClick={() => handleToggle(todo.id)}
                >
                    {todo.desc}
                </p>
            </li>
            <button
                className='todo__item__delete'
                onClick={() => handleDelete(todo.id)}
            >
                Borrar
            </button>
        </>
    )
}
