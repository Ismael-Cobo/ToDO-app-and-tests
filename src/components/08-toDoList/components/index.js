export { AddToDo } from "./AddToDo";
export { Header } from "./Header";
export { ListToDo } from "./ListToDo";
export { FooterToDo } from "./FooterToDo";