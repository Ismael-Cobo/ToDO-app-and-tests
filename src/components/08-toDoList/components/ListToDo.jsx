import { TodoListItem } from "./TodoListItem"

export const ListToDo = ({ todos, handleDelete, handleToggle }) => {

    return (
        <section className='todo__items'>
            <ul className='todo__items__list'>
                {
                    todos
                        ? todos.map(todo => (
                            <div
                                className='todo__item__container'
                                key={todo.id}
                            >
                                <TodoListItem
                                    todo={todo}
                                    handleDelete={handleDelete}
                                    handleToggle={handleToggle}
                                />
                            </div>
                        ))
                        : ''
                }
            </ul>
        </section>
    )
}