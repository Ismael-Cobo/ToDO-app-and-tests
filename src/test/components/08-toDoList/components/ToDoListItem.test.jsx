import { shallow } from "enzyme"
import { TodoListItem } from "../../../../components/08-toDoList/components/TodoListItem"
import { demoReducer } from "../../../fixtures/demoTodos"

describe('pruebas en el componente <ListToDo />', () => {

    const todo = demoReducer[0]
    const handleDelete = jest.fn()
    const handleToggle = jest.fn()


    const wrapper = shallow(
        <TodoListItem
            todo={todo}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
        />
    )

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de llamar la función handleDelete', () => {

        const button = wrapper.find('.todo__item__delete')

        button.simulate('click', { id: todo.id })

        expect(handleDelete).toHaveBeenCalledWith(todo.id)
        expect(handleDelete).toHaveBeenCalled()
    })

    test('debe de llamar la función handleTogle', () => {
        const text = wrapper.find('.todo__item__text')

        text.simulate('click', { id: todo.id })

        expect(handleToggle).toHaveBeenCalledWith(todo.id)
    })

    test('debe de mostrar el texto correctamente', () => {
        const text = wrapper.find('.todo__item__text')

        expect(text.text()).toBe(todo.desc)

    })

})