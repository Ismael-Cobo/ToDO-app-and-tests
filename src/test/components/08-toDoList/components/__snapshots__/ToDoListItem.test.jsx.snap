// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <ListToDo /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="todo__item "
  >
    <p
      className="todo__item__text"
      onClick={[Function]}
    >
      hacer esto
    </p>
  </li>
  <button
    className="todo__item__delete"
    onClick={[Function]}
  >
    Borrar
  </button>
</Fragment>
`;
